Refresh.GameServer/Database/GameDatabaseContext.Relations.cs

```diff
@@ -184,13 +161,7 @@ public void ClearQueue(GameUser user)
         => this._realm.All<RateLevelRelation>().FirstOrDefault(r => r.User == user && r.Level == level);
 
     [Pure]
-    public RatingType GetRatingByUser(GameLevel level, GameUser user)
-    {
-        RateLevelRelation? rating = this.GetRateRelationByUser(level, user);
-        if(rating == null) return RatingType.Neutral;
-            
-        return rating.RatingType;
-    }
+    public RatingType? GetRatingByUser(GameLevel level, GameUser user) => this.GetRateRelationByUser(level, user)?.RatingType;
```

Comment:
I prefer the old method. It's clear what's actually happening when the rating is `null` to the reader. You might know that the default `RatingType` is `Neutral`, but someone else might not.

Reply:
PSP/LBP1 ratings need to know the difference between "no rating" and "neutral rating", would i just call into `GetRateRelationByUser` directly for PSP/LBP1?

Reply:
Oh, I misunderstood the purpose of this refactor. Yeah, just do that.

Reply:
hmm, seems the only usage of `GetRatingByUser` is [here](https://github.com/LittleBigRefresh/Refresh/blob/abe8b392b748b76968a4de0bc79109032904c89d/Refresh.GameServer/Endpoints/Game/DataTypes/Response/GameLevelResponse.cs#L129) so using `GetRateRelationByUser` directly for LBP1/PSP would either double the database calls made or mean GetRatingByUser goes completely unused in the codebase

We should also probably encourage any potential future users of `GetRatingByUser` to be aware of the semantics of "neutral vs missing", since there *is* a difference now, perhaps an xmldoc explaining what null should mean in use, with my refactored implementation would be clear enough?

Reply:
I see. I think the nullable marker is enough to convey "the rating is missing", but it couldn't hurt to add xmldoc.